qutebrowser/mainwindow/mainwindow.py

```diff
@@ -483,9 +487,14 @@ def _connect_signals(self):
     @pyqtSlot(bool)
     def _on_fullscreen_requested(self, on):
         if on:
+            self._restore_fullscreen = self.isFullScreen()
             self.showFullScreen()
-        else:
+        elif not self._restore_fullscreen:
             self.showNormal()
+        else:
+            self._restore_fullscreen = self.isFullScreen()
```

Comment:
This way the state is updated whenever the `_on_fullscreen_requested` method is called, including when there are multiple tabs and I close a tab by middle-clicking on the tab bar. Switching from video tabs to normal tabs is still not perfect, but is there a reason why _not_ to do this?